### ISSUE TITLE
Regression: get should have a default value param

### ISSUE THREAD

**Issue by dmvdbrugge** [parable-php/getset#1]
`get($key, $default = null)`

Returning `$default` on not found.

**Comment by devvoh** [comment]
This was a specific choice. In Parable 1.x, which was php 5.6+, null coalesce was not available.

Before: `$value = $collection->get('key', 123);`
Now: `$value = $collection->get('key') ?? 123;`

Obviously I'm open to any arguments where this would not be a workable alternative, but I can't think of any.

**Comment by Thijs-Riezebeek** [comment]
I also prefer the latter, as it is explicit that it is your default in case of a `null` return, whereas the second parameter way of doing it doesn't make this clear. Using a language construct such as `??` for this, and even forcing it, is great imo.

**Comment by dmvdbrugge** [comment]
The edge case though, is if a value can be null. How do I know if it was not set, or if it was specifically set as null? Default param covers that case, while null-coalesce doesn't.

Also, nothing forbids you from doing null-coalesce, even with default param.

**Comment by devvoh** [comment]
I see what you mean there, since scenarios 'is not set' and 'is set to `null`' both result in `null` always.

The default param allows:

`$value = $collection->get('key', 123);` and it would be 123 when it's not set, `null` if set to `null`.

This isn't possible without a default param. The only way to approximate this behaviour is the following:

```
if ($collection->has('key')) {
    $value = $collection->get('key');
} else {
    $value = 123;
}
```

And since the entire point of GetSet is to be easy to use and to take the hassle of working with these data structures away, I'm adding back the default parameter. Thanks!

**Comment by dmvdbrugge** [comment]
You fixed this in 0e4e9da3f7136f0b0c4b71959a0d914454a06a84 (🎉) and you changed the readme. In that readme change, you also added the default param to `getAndRemove`, however in the actual code you forgot to add it.

**Comment by devvoh** [comment]
You're right. I planned on doing that but then realized that since `remove()` throws on non-existing values, it would never return the default anyway.

**Comment by dmvdbrugge** [comment]
Fixed the followup issue in 9edbfd254bd7ad26333233fe187e3cc7709b3d87
